src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -184,10 +184,26 @@ public Response addCluster(
     }
     Optional<Cluster> existingCluster = context.storage.getCluster(newCluster.getName());
     if (existingCluster.isPresent()) {
-      LOG.info("cluster already stored with this name: {}", existingCluster);
-      return Response.status(403)
-          .entity(String.format("cluster \"%s\" already exists", existingCluster.get().getName()))
-          .build();
+      LOG.info(
+          "Cluster already stored with this name: {}. Trying to updating the node list.",
+          existingCluster.get().getName());
+      Cluster updatedCluster = updateClusterSeeds(existingCluster.get(), seedHost.get());
+      if (updatedCluster.getSeedHosts().equals(existingCluster.get().getSeedHosts())) {
+        LOG.info("Node list of cluster {} is already up to date.", existingCluster.get().getName());
+        return Response.notModified()
+            .entity(
+                String.format(
+                    "Topology hasn't changed in cluster %s. No update was performed.",
+                    existingCluster.get().getName()))
+            .build();
+      } else {
+        LOG.info(
+            "Node list of cluster {} has been updated with the current topology.",
+            existingCluster.get().getName());
+        return Response.ok()
+            .entity(String.format("Updated cluster %s node list", existingCluster.get().getName()))
+            .build();
```

Comment:
it's odd behaviour that this rest endpoint can return 200, but in one situation return a string message, and in another situation return the ClusterStatus json…

This has been addressed in #401